### ISSUE TITLE
v2.1 user adoption is slower than historical data

### ISSUE THREAD

**Issue by Sdaswani** [mozilla-mobile/firefox-tv#872]
It's not sure what other data we are losing, so this may be a candidate for a 2.1.1 .

### Are users using the latest app?
Not according to this: https://sql.telemetry.mozilla.org/queries/52029#138894

### Investigate Reason for Uptake not 100%
Thanks to help from Frank, looks like latest Fire OS version isn't reporting some telemetry: https://sql.telemetry.mozilla.org/queries/52806#140500

**Comment by mcomella** [comment]
Looking at these graphs, it doesn't look like we're losing any users: adding together the number of users in 2.0.1 with 2.1 gives us the same number of overall users. A large number of users haven't upgraded yet but I assume that's part of the process of the Amazon store.

**Comment by Sdaswani** [comment]
Thanks @mcomella let me talk to Amazon, I thought by now 2.1 should be at 100%.

**Comment by mcomella** [comment]
To be explicit, the problem with this bug is not that our data is incorrect, it's that adoption of 2.1 is very slow compared to previous releases.

**Comment by Sdaswani** [comment]
Thanks @mcomella I’m checking with Amazon.

**Comment by liuche** [comment]
Amazon says that this might require a manual update from the user, so that they can accept the new permissions. They suggest sending a notification to prompt users to update.

**Comment by Sdaswani** [comment]
@bbinto so notifications are enabled by default, so we could push out a notification to increase the uptake rate.

[Blog post](https://developer.amazon.com/blogs/appstore/post/4cdd501a-932d-4ccb-97fa-dca6b17c659f/notifications-and-new-amazon-fire-tv-user-interface-puts-content-first.) discussing notifications: 

> All the details you need to get started with notification on Fire TV are outlined in our [documentation](https://developer.amazon.com/public/solutions/devices/fire-tv/docs/notifications-for-amazon-fire-tv?sc_channel=blog&sc_publisher=blog&sc_campaign=Launch_Fire-TV-UI&sc_assettype=launch&sc_team=us&sc_traffictype=organic&sc_country=world-wide&sc_segment=all&sc_itrackingcode=100011_us_blog). Notifications are enabled by default and images are supported, helping you create more engaging communication with your customers. All notifications will be shown in the dedicated Notification Center, where customers will be provided with visual cues to learn that they can expect to hear directly from the developers of their favorite content within the Notification Center.

**Comment by bbinto** [comment]
Ok, let's do it, all we need to provide is release notes bullets?


- Bug fixes 
- Updated Your Rights page 
- Improved cursor

How much eng effort is this, @Sdaswani & @liuche ?

**Comment by Sdaswani** [comment]
@bbinto I think this at least a Medium effort (2-3 days for one developer). We have to:

1. Create the notification
2. Build some tools to send the notification to the Amazon Push Service (need to find this documentation).
3. Test to make sure the whole delivery flow works.

Step 2 may turn out more difficult so it could be a 3-5 day (L) effort.

**Comment by bbinto** [comment]
@Sdaswani - will people who have not updated to 2.1 automatically not be updated to 3.0 either?

If that's the case, I feel we probably have to include this issue in our 3.0 release.

**Comment by Sdaswani** [comment]
The answer to that question is yes.

But note there is nothing to include with 3.0 AFAICT - we can send the notification any time, it shouldn’t be required to be in an app release.

**Comment by mcomella** [comment]
Looking at the specific [notification documentation](https://developer.amazon.com/docs/fire-tv/notifications.html?sc_channel=blog&sc_publisher=blog&sc_campaign=Launch_Fire-TV-UI&sc_assettype=launch&sc_team=us&sc_traffictype=organic&sc_country=world-wide&sc_segment=all&sc_itrackingcode=100011_us_blog#system-notifications), it looks like they already send out notifications when the app needs an update:

> Amazon Fire TV also provides notifications when your app has an update. These notifications aren't something you create with your app but rather are triggered by the Fire TV appstore client.
>
> Fire TV creates two types of app update messages. A "Required Update" message is a visual prompt on an app icon, indicating that there is a new update available.
>
> Another update message is presented to users though an on-device dialogue box. When users start a new session in your app or game, they are presented with the option to "Update Now" or "Launch without Updating" along with details describing what’s new in the update.

A few things I'm wondering about:
- Do these happen automatically (I assume so) or do we need to trigger them ourselves?
  - If these happen automatically, what benefit do we have to sending our own notifications? Perhaps these are new?
- The dialog update should happen when a new app session is started: is it that our users aren't leaving the app so aren't seeing the dialog? Do they dismiss it?
- The docs don't mention push, meaning we'd have to add code to send notifications, which users will be unable to get (without regressing our permission add). Am I missing something?

@Sdaswani Do you know more here? Since we've added another permission for the MediaSession API #930 (it's an Amazon permission though so I don't know if it'd have the same effect), I was thinking it might be a good time to send a notification to try to get everyone to the latest version all at once.

**Comment by Sdaswani** [comment]
- I believe they happen automatically, yes.
-- The benefit would be to remind them more, since we are seeing the update uptake is low.
- Your guess is as good as mine. You'd think they are nagged pretty constantly to update, so it's a wonder why they aren't updating faster. Maybe there is a second dialog that pops up before they can update, and that second dialog has a higher drop off rate. I'm going to ask Amazon if they can fill in more details.
- No, that's right, we neither have the push permission nor the backend infrastracture to push to folks who accepted the permission.

**Comment by mcomella** [comment]
I got a notification to update Silk on the Fire Stick: here's what I found out.
- I received a heads up display notification shortly after becoming active (ignored it)
- ^ This notification went into the "Settings" panel. After reading and ignoring it, it disappeared and the notification panel is empty (I can't read old notifications, it seems)
- I received a full screen dialog to update when I first opened the app. However, subsequent opens do not show the update dialog, even after force stop and restarting the device. **It seems like the docs are incorrect or there is a bug** about a notification being shown for each new app session. Perhaps it's time sensitive?
- After ignoring the dialog and ignoring the notification, there is no obvious indication of an update after this. In order to update, I need to manually go to the app's info page (kinda like their store page) in order to see the update and perform the update.

Maybe we'll give users a second round of notifications when we push v2.2 or subsequent versions?
